src/firefly/js/ui/FileUploadProcessor.jsx

```diff
@@ -372,7 +374,8 @@ function sendTableRequest(tableIndices, fileCacheKey, treatAsSpectrum, currentRe
         const {index} = parts[idx];
         const fileRoot= fileName?.split('.')?.[0] ?? fileName;
         const extMarker= getExtMarker(parts[idx], idx);
-        const title = parts.length > 1 ? `${fileRoot}:${extMarker}` : fileRoot;
+        const title = parts.length > 1 ? hasExtName(parts[idx]) ?
+            `${extMarker}:${fileRoot}` : `${fileRoot}:${extMarker}` : fileRoot;
```

Comment:
Since this is a nested ternary, as a personal preference I would suggest this small refactor for better readability: 

```
const title = parts.length > 1 
  ? (hasExtName(parts[idx]) 
      ? `${extMarker}:${fileRoot}` 
      : `${fileRoot}:${extMarker}`) 
  : fileRoot;
```